python/lsst/pipe/base/pipelineIR.py

```diff
@@ -87,6 +92,66 @@ def __eq__(self, other: "ContractIR"):
             return False
 
 
+@dataclass
+class LabeledSubset:
+    """Intermediate representation of named subset of task labels read from
+    a pipeline yaml file.
+    """
+    label: str
+    """The label used to identify the subset of task labels.
+    """
+    subset: Set[str]
+    """A set of task labels contained in this subset.
+    """
+    description: Optional[str]
+    """A description of what this sub set of tasks is intended to do
+    """
+
+    @staticmethod
+    def from_yaml(label: str, value: Union[List[str], dict]) -> LabeledSubset:
+        """This static method generates `LabeledSubset` objects given a properly
+        formatted object that as been created by a yaml loader.
```

Comment:
Docstring should [start with the imperative voice](https://developer.lsst.io/python/numpydoc.html#writing-summaries-for-functions-and-methods).